### ISSUE TITLE
"Builders Digging Holes" bug from last year seems to be back

### ISSUE THREAD

**Issue by Esheon** [ldtteam/minecolonies#5234]
Version: minecolonies-1.15.2-0.11.959-universal.jar
structurize-0.10.323-ALPHA.jar
Forge 31.2.9

I'm using a test version of my "EG Colonies" modpack, slightly ahead of the one available on CurseForge. There are other active mods, though none that should affect Minecolonies or Structurize.

### Expected behavior
Builder only digs out what is necessary to place the blocks for the hut he is building

### Actual behaviour
Builder digs out several levels below what is needed for the hut.

I updated to the latest version of Forge, Minecolonies and Structurize tonight, hoping it would fix another issue I was having. I decided to start a new game and colony, this time in Medieval Spruce. I placed the town hall block and the builder's hut block (using the building tool, not barehanded), but when the builder started work he just wanted shovels (he went through about half a dozen) and pickaxes (another half dozen). He dug to about 5 blocks below the surface of the entire footprint (not just where the level 1 builder would be, the entire area within the construction tape) before starting to build. When he started building he placed grass blocks (technically dirt blocks placed as grass blocks) only under the actual hut location, leaving a gap around the finished hut. I built a tavern next in the same world, and the builder acted in the same manner.

I wanted to make sure it wasn't an issue with the Medieval Spruce schematics, so I created a new world in superflat with the overworld preset. I placed the town hall and started four builder's huts: Medieval Oak, Medieval Dark Oak, Medieval Birch, and Caledonia (those are the only other builder's huts that have a basement). I supplied resources to those huts with the creative menu; I did not use the setting that makes builders not consume resources. All four builders began digging the entire footprint out until they were five blocks below the surface, and finished the builds in the same way as the previous world.

So at the very least, the issue seems to exist with all of the standard Medieval-style builder's huts (I didn't test the alternate styles) and the Medieval Spruce Tavern. I didn't have time to test other buildings tonight (I should have gone to bed an hour ago).

As an incidental aside, the issue I was hoping would be fixed by the update was the Medieval Oak Town Hall requesting one more door than it actually needed and never being able to finish. I didn't get around to seeing if that was fixed or not.

Screenshots 
The slightly larger building footprint is the Caledonia builder's hut, the others are Medieval style.
The glowstone blocks mark the edges of the schematic footprints.

This is 4 blocks down from the surface
![mcpartial](https://user-images.githubusercontent.com/64860892/84757071-a6b1a480-af91-11ea-8237-712df9f932e7.jpg)

They are building back up with grass blocks now
![mcbuildstart](https://user-images.githubusercontent.com/64860892/84757110-b92bde00-af91-11ea-8792-5f1688364b7e.jpg)

And a complete builder's hut with the gap around the edge
![mchutcomplete](https://user-images.githubusercontent.com/64860892/84757152-c779fa00-af91-11ea-8dfb-17a366c19f37.jpg)

I let it keep running while I was typing this out, and the three Medieval-style huts all ended up with the same gap around them. The Caledonia hut was normal, but it also takes up the whole footprint instead of just part of it.

### Steps to reproduce the problem
1. Create a world
2. Place the supply camp, town hall, and any of the above listed buildings
3. Tell the builder to start working
4. Watch the madness

### Logs
latest.log - https://gist.github.com/Esheon/d6869f4513e85c27293ce52f7d4cb09c
No crashlog, the game never crashed.

The linked "latest.log" file is from the session in which I took the above screenshots.

**Comment by Raycoms** [comment]
fixed
